code/datums/quirks/neutral_quirks/transhumanist.dm

```diff
@@ -0,0 +1,147 @@
+	RegisterSignal(quirk_holder, COMSIG_CARBON_POST_REMOVE_LIMB, PROC_REF(calculate_bodypart_score))
+	RegisterSignal(quirk_holder, COMSIG_CARBON_GAIN_ORGAN, PROC_REF(calculate_bodypart_score))
+	RegisterSignal(quirk_holder, COMSIG_CARBON_LOSE_ORGAN, PROC_REF(calculate_bodypart_score))
+	calculate_bodypart_score()
+
+/datum/quirk/transhumanist/remove()
+	UnregisterSignal(quirk_holder, list(COMSIG_CARBON_REMOVE_LIMB, COMSIG_CARBON_ATTACH_LIMB))
+
+/datum/quirk/transhumanist/proc/calculate_bodypart_score()
+	SIGNAL_HANDLER
+	var/mob/living/carbon/human/human_holder = quirk_holder
+	var/organic_bodytypes = 0
+	var/silicon_bodytypes = 0
+	var/other_bodytypes = FALSE
+	for(var/obj/item/bodypart/part as anything in human_holder.bodyparts)
+		if(part.bodytype & BODYTYPE_ROBOTIC)
+			silicon_bodytypes += 1
+		else if(part.bodytype & BODYTYPE_ORGANIC)
+			organic_bodytypes += 0.1
+		else
+			other_bodytypes = TRUE
+
+	for(var/obj/item/organ/organ as anything in human_holder.organs)
+		if(organ.organ_flags & ORGAN_ROBOTIC)
+			silicon_bodytypes += 0.25
+		else if(organ.organ_flags & ORGAN_ORGANIC)
+			organic_bodytypes += 0.02
+
+	if(!other_bodytypes)
+		if(organic_bodytypes == 0)
+			quirk_holder.add_mood_event(MOOD_CATEGORY_TRANSHUMANIST_BODYPART, /datum/mood_event/completely_robotic)
+			return
+		else if(silicon_bodytypes == 0)
+			quirk_holder.add_mood_event(MOOD_CATEGORY_TRANSHUMANIST_BODYPART, /datum/mood_event/completely_organic)
+			return
+	else if(silicon_bodytypes == 0 && organic_bodytypes == 0)
+		quirk_holder.clear_mood_event(MOOD_CATEGORY_TRANSHUMANIST_BODYPART)
+		return
+
+	var/bodypart_score = silicon_bodytypes - organic_bodytypes
+	switch(bodypart_score)
+		if(3 to INFINITY)
+			quirk_holder.add_mood_event(MOOD_CATEGORY_TRANSHUMANIST_BODYPART, /datum/mood_event/very_robotic)
+		if(0 to 3)
+			quirk_holder.add_mood_event(MOOD_CATEGORY_TRANSHUMANIST_BODYPART, /datum/mood_event/balanced_robotic)
+		if(-INFINITY to 0)
+			quirk_holder.add_mood_event(MOOD_CATEGORY_TRANSHUMANIST_BODYPART, /datum/mood_event/very_organic)
+
+
+/datum/quirk/transhumanist/add_unique(client/client_source)
+	var/limb_type = GLOB.limb_choice_transhuman[client_source?.prefs?.read_preference(/datum/preference/choiced/prosthetic)]
+	if(isnull(limb_type))  //Client gone or they chose a random prosthetic
+		limb_type = GLOB.limb_choice_transhuman[pick(GLOB.limb_choice_transhuman)]
+
+	var/mob/living/carbon/human/human_holder = quirk_holder
+
+	var/obj/item/bodypart/new_part = new limb_type()
+	var/obj/item/bodypart/current_zone = human_holder.get_bodypart(new_part.body_zone)
+
+	slot_string = "[new_part.plaintext_zone]"
+	old_limb = human_holder.return_and_replace_bodypart(new_part, special = TRUE)
+
+/datum/quirk/transhumanist/post_add()
+	if(slot_string)
+		to_chat(quirk_holder, span_boldannounce("Your [slot_string] has been replaced with a robot arm. You need to use a welding tool and cables to repair it, instead of sutures and regenerative meshes."))
```

Comment:
```suggestion
		to_chat(quirk_holder, span_boldannounce("Your [slot_string] has been replaced with a robotic limb. You need to use a welding tool and cables to repair it, instead of sutures and regenerative meshes."))
```